### PR TITLE
Initial categorization of release notes

### PR DESCRIPTION
**Pull request by trbromley**

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -63,31 +84,26 @@
...
+* Added `ops.functions.assert_valid` for checking if an `Operator` class is defined correctly.
+  [(#4764)](https://github.com/PennyLaneAI/pennylane/pull/4764)
+
+* `GlobalPhase` now decomposes to nothing, in case devices do not support global phases.
+  [(#4855)](https://github.com/PennyLaneAI/pennylane/pull/4855)
 
-* `has_matrix` will now automatically be `True` if `Operator.matrix` is overridden, even if
+* Custom operations can now provide their matrix directly through the `Operator.matrix()` method
+  without requiring `Operator.compute_matrix()` to be provided. Behind the scenes,
+  `has_matrix` will now automatically be `True` if `Operator.matrix` is overridden, even if
```
Comment (trbromley) [inline doc/releases/changelog-dev.md:100]: This was updated a bit more to emphasize the user-facing consequence.

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -192,12 +208,12 @@
   integer power.
   [(#4827)](https://github.com/PennyLaneAI/pennylane/pull/4827)
 
-
 <h3>Contributors ✍️</h3>
 
 This release contains contributions from (in alphabetical order):
 
 Guillermo Alonso,
+Thomas Bromley,
```
Comment (trbromley) [inline doc/releases/changelog-dev.md:216]: @isaacdevlugt is your PR still to be merged?

**Review by mlxd — CHANGES_REQUESTED** [review]
One quick change, but nothing else from my side

File: doc/releases/changelog-dev.md (hunk 1 above)
Comment (albi3ro) [reply doc/releases/changelog-dev.md:104]: I'm ok with the change, but we never really required `Operator.compute_matrix` to be overridden. You just then had to manually update `has_matrix` to get device support.

**Review by albi3ro — APPROVED** [review]

File: doc/releases/changelog-dev.md (hunk 1 above)
Comment (trbromley) [reply doc/releases/changelog-dev.md:104]:
```suggestion
* Custom operations can now provide their matrix directly through the `Operator.matrix()` method
  without needing to update the `has_matrix` property. `has_matrix` will now automatically be
  `True` if `Operator.matrix` is overridden, even if
```
@albi3ro does this look better?

**Review by mlxd — APPROVED** [review]

**Pull request merged**